commons/src/main/java/org/infinispan/commons/util/EmptyQueue.java

```diff
@@ -0,0 +1,104 @@
+package org.infinispan.commons.util;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Queue;
+
+import static java.util.Collections.emptyIterator;
+
+/**
+ * @author gustavonalle
+ * @since 7.2
+ */
+public final class EmptyQueue<T> implements Queue<T> {
```

Comment:
This class should probably be hidden behind ```InfinispanCollections.emptyQueue()``` instead of being public, similar to ```InfinispanCollections.emptyList/Map/Set()```.